### DISCUSSION TITLE
Can Supabase slow down my server's memory? A question about Supabase Policy with Max Memory utilization per day.

### DISCUSSION THREAD

**Discussion by emisempo1** [supabase/supabase#30738] [Questions]
Supabase say to me : " Your project is about to deplete its Disk IO Budget, and may become unresponsive once fully exhausted
You will need to optimize your performance or upgrade your compute. Check the usage page for more recent and detailed statistics."  and always in the console of chrome i see 'canceling statement due to statement timeout' 
![LENTITUD SUPABASE](https://github.com/user-attachments/assets/a0dd063c-7f69-47d2-9886-994f13d9c8fc)
in the past this api reurns me 200 OK with the same query ,I have optimized my frontend to consume less resources and i never exceed the maximum memory usage per day but in my opinion supabase slow down my server's memory , this is possible ? Hug for all .

**Comment by GaryAustin1** [comment]
Not clear what your supabase instance has to do with your server memory.   A hosted instance does not use your apps server memory.  The statement timeout seems to be saying your query is exceeding the Postgres timeout either because you are overloading the Supabase database server with too much to do (deplete disk I/O is a clue on that) or that query is too complicated or on a large table.

This has more info:  https://github.com/orgs/supabase/discussions/14256
